src/forms/FormControl/FormControl.test.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+
+import {FormControl} from './FormControl'
+import {TextInput} from '../TextInput'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+expect.extend(toHaveNoViolations)
+
+describe('FormControl', () => {
+  const mockFormControlLabel = 'My label'
+  const mockFormControlId = 'my-input-id'
+  const mockTestId = 'my-test-id'
+
+  afterEach(cleanup)
+
+  it('renders a text input form control correctly into the document', () => {
+    const {getByRole, getByLabelText} = render(
+      <FormControl>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+
+    const labelEl = getByLabelText(mockFormControlLabel)
+    const inputEl = getByRole('textbox')
+
+    expect(labelEl).toBeInTheDocument()
+    expect(inputEl).toBeInTheDocument()
+  })
+
+  it('has no a11y violations', async () => {
+    const {container} = render(
+      <FormControl>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+    const results = await axe(container)
+
+    expect(results).toHaveNoViolations()
+  })
+
+  it('generates a unique ID and forwards to relevant children', async () => {
+    const {container, getByRole, getByTestId} = render(
+      <FormControl id={mockFormControlId} data-testid={mockTestId}>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+    const labelEl = container.querySelector('label')
+    const labelForValue = labelEl?.getAttribute('for')
+
+    const inputEl = getByRole('textbox')
+    const rootEl = getByTestId(mockTestId)
+
+    const inputIdValue = inputEl.getAttribute('id')
+    const inputNameValue = inputEl.getAttribute('name')
+
+    expect(inputIdValue).toEqual(labelForValue)
+    expect(inputNameValue).toEqual(labelForValue)
+    expect(rootEl.getAttribute('id')).toBe(mockFormControlId)
+  })
+
+  it('can forward an ID override to relevant children', async () => {
+    const {container, getByTestId, getByRole} = render(
+      <FormControl id={mockFormControlId} data-testid={mockTestId}>
+        <FormControl.Label>{mockFormControlLabel}</FormControl.Label>
+        <TextInput />
+      </FormControl>
+    )
+
+    const labelEl = container.querySelector('label')
+    const labelForValue = labelEl?.getAttribute('for')
+
+    const inputEl = getByRole('textbox')
+    const rootEl = getByTestId(mockTestId)
+
+    const inputIdValue = inputEl.getAttribute('id')
+    const inputNameValue = inputEl.getAttribute('name')
+
+    expect(inputIdValue).toEqual(labelForValue)
+    expect(inputNameValue).toEqual(labelForValue)
+    expect(rootEl.getAttribute('id')).toBe(mockFormControlId)
+  })
+
+  it('applies error state to the inputs and form validation', async () => {
+    const expectedMessage = 'My error message'
```

Comment:
The validation messages should be set to `--brand-text-size-200`

Reply:
Fixed in https://github.com/primer/brand/pull/69/commits/987d4c0e8d1456500b3a0f0dbf30c12e4eb3b67c